### PR TITLE
Add the ability to get the number of objects in mempack

### PR DESCRIPTION
**Pull request by nightelf3**
~Implement git_mempack_empty, which returns 1 if the mempack is empty and 0 otherwise.~
Implement git_mempack_object_count, which returns the number of objects in mempack and -1 on error

### CODE REVIEW

File: include/git2/sys/mempack.h [hunk 1]
```diff
@@ -101,6 +101,14 @@ GIT_EXTERN(int) git_mempack_dump(git_buf *pack, git_repository *repo, git_odb_ba
  */
 GIT_EXTERN(int) git_mempack_reset(git_odb_backend *backend);
 
+/**
+ * Checks if mempack is empty
+ *
+ * @param backend The mempack backend
+ * @return 1 if the repository is empty, 0 if it isn't
```
Comment (ethomson) [inline include/git2/sys/mempack.h:108]: and `-1` on error

File: src/libgit2/odb_mempack.c [hunk 2]
```diff
@@ -209,3 +209,14 @@ int git_mempack_new(git_odb_backend **out)
 	*out = (git_odb_backend *)db;
 	return 0;
 }
+
+int git_mempack_empty(git_odb_backend *_backend)
+{
+	struct memory_packer_db *db = (struct memory_packer_db *)_backend;
+	GIT_ASSERT_ARG(_backend);
+
+	if (git_odb_mempack_oidmap_size(&db->objects) || db->commits.size)
```
Comment (ethomson) [inline src/libgit2/odb_mempack.c:218]: I'm curious about the logical or here - is there ever a case where these two would disagree?

**Review by ethomson — COMMENTED** [review]
Thanks for the PR - this seems like a reasonable addition.

Thinking about the future - would we ever want to know the number of items in the mempack? If so, then maybe we should move in that direction now. I'm not sure that I want an API for `git_mempack_empty` and then later add a `git_mempack_count` or something, since that would make the `_empty` API superfluous. 🤔

File: include/git2/sys/mempack.h (hunk 1 above)
Comment (nightelf3) [reply include/git2/sys/mempack.h]: Done

File: src/libgit2/odb_mempack.c (hunk 2 above)
Comment (nightelf3) [reply src/libgit2/odb_mempack.c]: Done, no need to check number of commits, thank you!

**Comment by nightelf3** [comment]
I like the idea of getting the number of objects in mempack instead of `_empty()`.
PR is updated.

**Comment by ethomson** [comment]
Thanks again - I moved the count to an `out` param and manually merged this. 🙏

**Pull request closed without merging**